### PR TITLE
Allow oci memory store to match containers by id prefix or by name

### PR DESCRIPTION
**Pull request by 14rcole**
Useful for commands such as `kpod stop` or `kpod logs`

### CODE REVIEW

File: oci/container.go [hunk 1]
```diff
@@ -184,3 +184,7 @@ func (c *Container) State() *ContainerState {
 	defer c.opLock.Unlock()
 	return c.state
 }
+
+func (c *Container) partialIDMatch(id string) bool {
```
Comment (mrunalp) [inline oci/container.go]: We have a data structure for this. Take a look at how we look up containers in getContainerFromRequest.

File: oci/memory_store.go [hunk 2]
```diff
@@ -24,18 +24,26 @@ func (c *memoryStore) Add(id string, cont *Container) {
 }
 
 // Get returns a container from the store by id.
-func (c *memoryStore) Get(id string) *Container {
+func (c *memoryStore) Get(idOrName string) *Container {
```
Comment (mheon) [inline oci/memory_store.go]: Why are we mutating existing functions? I would strongly prefer that we retain Get()'s current semantics.

I also don't feel like functions are appropriate for MemoryStore. Make them act on ContainerServer, so you can use the Name and ID indexes for fast lookups.

File: oci/container.go (hunk 1 above)
Comment (14rcole) [inline oci/container.go]: Will do

File: oci/memory_store.go (hunk 2 above)
Comment (14rcole) [inline oci/memory_store.go]: Okay, I reverted this and moved the server's getContainerFromRequest() function to ContainerServer

**Comment by 14rcole** [comment]
@mrunalp @mheon This should be fixed now

**Comment by mheon** [comment]
@14rcole Does this still do everything you needed? The old version matched on name and (optionally partial) ID, this version only matches on ID. I wouldn't be opposed to adding generic LookupContainer() and LookupSandbox() functions that can accept either.

**Comment by 14rcole** [comment]
I didn't realize this only matched on id.  I'll add the generic functions to support matching on name as well.  We need to be able to look up containers by name, otherwise there's no point in allowing users to give containers easy to remember names upon creation

**Comment by 14rcole** [comment]
@mrunalp how does this look?

**Comment by mheon** [comment]
Changes look good on my end, LGTM

**Comment by mrunalp** [comment]
LGTM

**Comment by runcom** [comment]
LGTM

**Pull request merged**